game-core/src/main/java/games/strategy/triplea/image/UnitImageFactory.java

```diff
@@ -156,14 +156,14 @@ public Image getImage(final UnitCategory unit) {
     final Optional<URL> imageLocation = getBaseImageUrl(baseImageName, id);
     Image image = null;
     if (imageLocation.isPresent()) {
-      image = Toolkit.getDefaultToolkit().getImage(getBaseImageUrl(baseImageName, id).get());
+      image = Toolkit.getDefaultToolkit().getImage(imageLocation.get());
       Util.ensureImageLoaded(image);
       if (needToTransformImage(id, type, mapData)) {
         image = convertToBufferedImage(image);
-        if (mapData.getUnitColor(id.getName()).isPresent()) {
-          final Color color = mapData.getUnitColor(id.getName()).get();
+        final Optional<Color> unitColor = mapData.getUnitColor(id.getName());
```

Comment:
nit, FWIW, AFAIK, canonical optional usage could have this be inlined:
mapData.getUnitColor(id.getName()).ifPresent(
   unitColor -> {
          final int brightness = mapData.getUnitBrightness(id.getName());
          ImageTransformer.colorize(unitColor.get(), brightness, (BufferedImage) image);
});

Reply:
This doesn't work because image is not final. We could restructure the code more (for example, early return on line 158, but I've left as is for now.

Reply:
Ah, okay, thanks for clarifying :+1: